R/PipeOpTuneThreshold.R

```diff
@@ -0,0 +1,98 @@
+#' @title PipeOpTuneThreshold
+#'
+#' @usage NULL
+#' @name mlr_pipeops_tunethreshold
+#' @format [`R6Class`] object inheriting from [`PipeOpTaskPreproc`]/[`PipeOp`].
+#'
+#' @description
+#' Tunes optimal probability thresholds over different [`PredictionClassif`]s.
+#' Learner's `predict_type` `"prob"` is required.
+#' Thresholds for each learner are optimized using (GenSA)[GenSA::GenSA].
+#' Returns a single [`PredictionClassif`].
+#' As a default, optimizes the miss-classification error [`MeasureClassif`].
+#' This PipeOp should be used in conjunction with [`PipeOpLearnerCV`] in order to
+#' optimize thresholds of predictions.
+#'
+#' @section Construction:
+#' ```
+#' * `PipeOpTuneThreshold$new(id, param_vals = list)` \cr
+#'   (`character(1)`, `list`) -> `self` \cr
+#' ```
+#' * `id` :: `character(1)`\cr
+#'   Identifier of resulting object.
+#' * `param_vals` :: named `list`\cr
+#'   List of hyperparameter settings, overwriting the hyperparameter settings
+#'   that would otherwise be set during construction. Default `list()`.
+#'
+#' @section Input and Output Channels:
+#' Input and output channels are inherited from [`PipeOpPredPostproc`].
+#'
+#' @section State:
+#' The `$state` is a named `list` with elements
+#' * `thresholds` :: `character` learned thresholds
+#'
+#' @section Parameters:
```

Comment:
``#' The parameters are the parameters inherited from [`PipeOpPredPostproc`], as well as:``